lib/chef/knife.rb

```diff
@@ -265,7 +225,13 @@ def self.load_deps
     # Error out and print usage. probably because the arguments given by the
     # user could not be resolved to a subcommand.
     def self.subcommand_not_found!(args)
-      ui.fatal("Cannot find sub command for: '#{args.join(' ')}'")
+      ui.fatal("Cannot find subcommand for: '#{args.join(' ')}'")
+
+      # Mention rehash when the subcommands cache(plugin_manifest.json) is used
+      if subcommand_loader.is_a?(Chef::Knife::SubcommandLoader::HashedCommandLoader) ||
+         subcommand_loader.is_a?(Chef::Knife::SubcommandLoader::CustomManifestLoader)
+        ui.info("If this is a recently installed plugin, please rehash to update the subcommands cache.")
```

Comment:
Done in https://github.com/chef/chef/commit/3bb418a2f2e8312c3373e54de58a6808f1e64911
Thanks for the review and feedback Dan